documentation/design-docs/eventcounters.md

```diff
@@ -0,0 +1,306 @@
+
+First, the EventSource that produces the counter value needs to be enabled. To do this, you can override the `OnEventSourceCreated` method to get a notification when an EventSource is created, and if this is the correct EventSource with your EventCounters, then you can call Enable on it. Here is an example of such override:
+
+```cs
+protected override void OnEventSourceCreated(EventSource source)
+{
+    if (source.Name.Equals("System.Runtime"))
+    {
+        Dictionary<string, string> refreshInterval = new Dictionary<string, string>()
+        {
+            { "EventCounterIntervalSec", "1" }
+        };
+        EnableEvents(source, 1, 1, refreshInterval);
+    }
+}
+```
+
+#### Sample Code
+
+This is a sample `EventListener` class that simply prints out all the counter names and values from a the .NET runtime's EventSource for publishing its internal counters (`System.Runtime`) at some interval.
+
+```cs
+public class SimpleEventListener : EventListener
+{        
+    private readonly EventLevel _level = EventLevel.Verbose;
+
+    public int EventCount { get; private set; } = 0;
+
+    private int _intervalSec;
+
+    public SimpleEventListener(int intervalSec)
+    {
+        _intervalSec = intervalSec;
+    }
+
+
+    protected override void OnEventSourceCreated(EventSource source)
+    {
+        if (source.Name.Equals("System.Runtime"))
+        {
+            Dictionary<string, string> refreshInterval = new Dictionary<string, string>() 
+            {
+                { "EventCounterIntervalSec", _intervalSec.ToString() }
+            };
+            EnableEvents(source, _level, (EventKeywords)(-1), refreshInterval);
+        }
+    }
+
+    private (string Name, string Value) getRelevantMetric(IDictionary<string, object> eventPayload)
+    {
+        string counterName = "";
+        string counterMean = "";
+        string counterIncrement = "";
+        bool isIncrement = false;
+
+        foreach ( KeyValuePair<string, object> payload in eventPayload )
+        {
+            string key = payload.Key;
+            string val = payload.Value.ToString();
+
+            if (key.Equals("Name"))
+            {
+                counterName = val;
+            }
+            else if (key.Equals("Mean"))
+            {
+                counterMean = val;
+            }
+            else if (key.Equals("Increment"))
+            {
+                counterIncrement = val;
+                isIncrement = true;
+            }
+        }
+
+        if (isIncrement)
+        {
+            return (counterName, counterIncrement);
+        }
+        else
+        {
+            return (counterName, counterMean);
+        }
+    }
+
+    protected override void OnEventWritten(EventWrittenEventArgs eventData)
+    {
+        for (int i = 0; i < eventData.Payload.Count; i++)
```

Comment:
Is it necessary to search every item in the payload? This appears too open-ended.